### ISSUE TITLE
Gateway throws error when federated schema contains Subscription type

### ISSUE THREAD

**Issue by jsangilve** [apollographql/federation#426]
- Package: @apollo/gateway (but the error is caused by @apollo/federation)
- Version: since @apollo/federation v0.9.1
- Last known working version: v0.6.x

When a federated service's schema contains a `Subscription` type and it's loaded by the Apollo Gateway, the following error is returned:

```
GraphQLSchemaValidationError: [accounts] Subscription -> `Subscription` is an extension type, but `Subscription` is not defined
```

I understand Subscriptions aren't supported by Federation yet, but having them in the downstream service shouldn't affect the creation of the federated graph.

While debugging the problem, I realized the Gateway was processing the Subscription as an extension type (the same way it does with Query/Mutation), but it's not adding an empty definition type for `Subscription`.

To reproduce it, just run the *federation demo* adding a `Subscription` to the `accounts `service:

```typescript
const typeDefs = gql`
  extend type Query {
    me: User
  }

  type User @key(fields: "id") {
    id: ID!
    name: String
    username: String
  }

  type Subscription {
    somethingHappened(input: String!): String!
  }
`;
```

Fork including the change above: https://github.com/jsangilve/federation-demo/blob/wip/example_with_subscription/services/accounts/index.js

**Comment by tot-ra** [comment]
We encountered same issue with Subscriptions.

@abernix our use-case is that we use federation for queries & mutations, while subscriptions are served by separate endpoint. But we still want to serve combined schema to the playground (graphiql)

**Comment by abernix** [comment]
@tot-ra We understand and we continue to consider improving the options we offer for employing Subscriptions and taking advantage of Federation.  Today, our only concrete suggestion is offered [in this blog post](https://www.apollographql.com/blog/backend/federation/using-subscriptions-with-your-federated-data-graph/).

**Comment by tot-ra** [comment]
@abernix yes, we've implemented same thing Mandi is doing. It doesn't solve the problem with schema validation though, because she is not running `composeAndValidate`. She is just merging gateway schema with built-in subscriptions https://github.com/apollosolutions/federation-subscription-tools/blob/main/src/utils/schema.js#L27

What we want is to validate & register subscription schema in our [schema-registry](https://github.com/pipedrive/graphql-schema-registry) - for audit, naming collisions & codestyle checks. But since we rely on `composeAndValidate` it blocks us.

**Comment by michjak-szymanski** [comment]
Hi. 
We are using Managed federation and we also need Subscriptions in our federated graphs. I've managed to implement them on our Apollo Gateway by following this tutorial https://www.apollographql.com/blog/backend/federation/using-subscriptions-with-your-federated-data-graph/ and it works.

However the problem is we cannot publish a subgraph that contains Subscription type into Apollo Studio (registry) because of following error we get from Rover:

```
Checking the proposed schema for subgraph ident against linus-digital-finance-muqldr@production
error[E029]: Encountered 1 build error while trying to build subgraph "ident" into supergraph "linus-digital-finance-muqldr@production".

Caused by:
    Encountered 1 build error while trying to build the supergraph.
    EXTENSION_WITH_NO_BASE: [ident] Subscription -> `Subscription` is an extension type, but `Subscription` is not defined in any service
    
        The changes in the schema you proposed for subgraph ident are incompatible with supergraph linus-digital-finance-muqldr@production. See https://www.apollographql.com/docs/federation/errors/ for more information on resolving build errors.

```

I've investigated the Rover code and it seems the error originates from `https://graphql.api.apollographql.com/api/graphq` which is beyond our control :)

Is there a way to bypass this error while still being able to have Subscription in **federated schema** on Apollo Studio?

**Comment by sammysaglam** [comment]
having waited for subscriptions support on federation  for a while now ⏳ , i was instead able to put together a small library implementing federation with subscriptions (the way we'd hoped for, without a separate service just for subscriptions). was able to get it working using schema-stitching & graphql-tools, it's basically just transforming federation SDL into schema-stitching SDL and using websockets across the gateway & subgraphs:
https://github.com/sammysaglam/federation-with-subscriptions

hope that helps

**Comment by jukben** [comment]
> What we want is to validate & register subscription schema in our [schema-registry](https://github.com/pipedrive/graphql-schema-registry) - for audit, naming collisions & codestyle checks.

I just second @tot-ra, we have exactly the same use case. Would you mind sharing how you have overcome it? 🙏

**Comment by jukben** [comment]
I just second to @tot-ra we have exactly the same use case with the difference we use Rover / Apollo Studio.

> What we want is to validate & register subscription schema in our schema registry.

Question to you @tot-ra, have you been able to overcome it in your solution? Looks pretty dope! 🙏

**Comment by min-proj** [comment]
> I get this error without subscription. I'm at the first stage of debugging this so I can't provide much more info.
> 
> It only occurs in production (I can't reproduce locally yet), with pm2 running multiple instances in cluster mode.
> 
> > '[tracker] Registration -> `Registration` is an extension type, but `Registration` is not defined in any service'

Did you resolve this issue? 
I am currently the same
